src/ppx_deriving_yojson.cppo.ml

```diff
@@ -684,12 +701,14 @@ let desu_sig_of_type ~options ~path type_decl =
       ]))
     in
     [mod_; of_yojson]
-  | _ -> [of_yojson; of_yojson_exn]
+  | _ ->
+    [of_yojson]
+    @ (if not want_exn then [] else [of_yojson_exn])
```

Comment:
only nit here and below would be to use `::` instead of `@` but the way it is right now probably makes it more concise...

Reply:
I think `(@)` is better here: the datatype with the right granularity for the problem domain is really "lists of structure items": an option may want to enable or disable the production of of two items at once, not just one (for example: a declaration and its `let _ = ...` warning-disabler), so it is sensible to compose these things by using `@` as the central operation.